djautotask/sync.py

```diff
@@ -476,9 +481,31 @@ def _assign_field_data(self, instance, object_data):
         return instance
 
 
-class ProjectSynchronizer(Synchronizer):
+class FilterProjectStatusMixin:
+
+    def fetch_records(self, query, results):
+        active_object_ids = self.get_active_ids()
+
+        for record in self.at_api_client.query(query):
+
+            object_id = getattr(record, self.lookup_field)
+            if active_object_ids and object_id:
+
+                if object_id not in active_object_ids:
+                    logger.info(
+                        'Project with ID: {} is set to an inactive status. '
+                        'Skipping this {}.'.format(
+                            object_id, self.model_class.__name__)
+                    )
+                    continue
+
+            self.persist_record(record, results)
+
+
+class ProjectSynchronizer(FilterProjectStatusMixin, Synchronizer):
     model_class = models.Project
     last_updated_field = 'LastActivityDateTime'
+    lookup_field = 'Status'
```

Comment:
Lookup field is a little to generic for this I think. I suggest changing this to relate to filtering, or something else that makes sense. Just a little more specific.

Reply:
Fixed.